packages/jellyfish-api-core/__tests__/category/account/withdrawFutureSwap.test.ts

```diff
@@ -0,0 +1,1209 @@
+      const withdrawFutureSwap: FutureSwap = {
+        address: withdrawAddress,
+        amount: `${withdrawAmount.toFixed(8)}@DUSD`,
+        destination: 'TSLA'
+      }
+      const result = testing.rpc.account.withdrawFutureSwap(withdrawFutureSwap)
+      await expect(result).rejects.toThrow(RpcApiError)
+      await expect(result).rejects.toThrow('RpcApiError: \': Amount out of range\', code: -3, method: withdrawfutureswap')
+    }
+
+    // withdraw after setting ${idTSLA}/dfip2203 to false
+    {
+      const idTSLA = await testing.token.getTokenId('TSLA')
+      await testing.rpc.masternode.setGov({ [attributeKey]: { [`v0/token/${idTSLA}/dfip2203`]: 'false' } })
+
+      const withdrawFutureSwap: FutureSwap = {
+        address: tslaAddress,
+        amount: `${swapAmount.toFixed(8)}@DUSD`,
+        destination: 'TSLA'
+      }
+      const result = testing.rpc.account.withdrawFutureSwap(withdrawFutureSwap)
+      await expect(result).rejects.toThrow(RpcApiError)
+      await expect(result).rejects.toThrow('RpcApiError: \'DFIP2203Tx: DFIP2203 currently disabled for token 2 (code 16)\', code: -26, method: withdrawfutureswap')
+
+      await testing.rpc.masternode.setGov({ [attributeKey]: { [`v0/token/${idTSLA}/dfip2203`]: 'true' } })
+    }
+
+    // withdraw after setting the dfip2203/active to false
+    {
+      await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'false' } })
+
+      const withdrawFutureSwap: FutureSwap = {
+        address: tslaAddress,
+        amount: `${swapAmount.toFixed(8)}@DUSD`,
+        destination: 'TSLA'
+      }
+      const result = testing.rpc.account.withdrawFutureSwap(withdrawFutureSwap)
+      await expect(result).rejects.toThrow(RpcApiError)
+      await expect(result).rejects.toThrow('RpcApiError: \'DFIP2203Tx: DFIP2203 not currently active (code 16)\', code: -26, method: withdrawfutureswap')
+
+      await testing.rpc.masternode.setGov({ [attributeKey]: { 'v0/params/dfip2203/active': 'true' } })
+    }
+
+    // verify that all happened before settle block
+    const currentBlock = await testing.rpc.blockchain.getBlockCount()
+    expect(currentBlock).toBeLessThan(nextSettleBlock)
+  })
+
+  it('Should not withdrawFutureSwap futureswap dtoken to dusd - at/after the settle block', async () => {
+    const swapAmount = 1
+    const nextSettleBlock = await testing.container.call('getfutureswapblock', [])
+    const tslaAddress = await testing.generateAddress()
+    await testing.rpc.account.accountToAccount(collateralAddress, { [tslaAddress]: `${swapAmount}@TSLA` })
+    await testing.generate(1)
+
+    const fswap: FutureSwap = {
+      address: tslaAddress,
+      amount: `${swapAmount.toFixed(8)}@TSLA`
+    }
+    await testing.rpc.account.futureSwap(fswap)
+    await testing.generate(1)
+
+    // move to next settle block
+    await testing.generate(nextSettleBlock - await testing.rpc.blockchain.getBlockCount())
```

Comment:
you already moving to next settle block here. -1 from this if you want to withdraw at settle block -> https://github.com/DeFiCh/jellyfish/pull/1323/files#r847048405